lib/Sema/SemaExpr.cpp

```diff
@@ -11935,7 +11935,20 @@ ExprResult Sema::ActOnCountBoundsExpr(SourceLocation BoundsKWLoc,
                                       CountBoundsExpr::Kind Kind,
                                       Expr *CountExpr,
                                       SourceLocation RParenLoc) {
-  return new (Context) CountBoundsExpr(Kind, CountExpr, BoundsKWLoc, RParenLoc);
+  ExprResult Result = UsualUnaryConversions(CountExpr);
+  if (Result.isInvalid())
+    return ExprError();
+  Expr *PromotedCountExpr = Result.get();
+  QualType ResultType = PromotedCountExpr->getType();
+  if (!ResultType->isIntegerType()) {
```

Comment:
>isIntegerType [](start = 19, length = 13)

Do we also need to confirm it's no larger than a pointer-sized integer (a.k.a. size_t or ptrdiff_t or something like that)?  Also, are requiring it to be signed or unsigned?  I don't recall what the spec says.